### DISCUSSION TITLE
How can I create a multi-language PDF (LaTeX) in LuaLaTeX?

### DISCUSSION THREAD

**Discussion by normansimonr** [quarto-dev/quarto-cli#11507] [Q&A]
### Description

I am writing a multi-language PDF book using XeLaTeX as the engine. The book combines different scripts (Latin, Hebrew, Greek) and has some complex formatting. So far XeLaTeX is working fine, but the output is not exactly perfect because there are horizontal spacing issues (words are sometimes separated by too much space). I believe this can be solved using the microtype package, but it only works at its best in LuaLaTeX.

When I change my enging to LuaLaTeX I get this error:

`ERROR: 
compilation failed- missing packages (automatic installed disabled)
LaTeX Error: Option clash for package babel.`

It [seems](https://quarto.org/docs/reference/formats/pdf.html#language) Quarto does not support Babel and LuaLaTeX concurrently (it does support either Babel + XeLaTeX or LuaLaTeX alone, but not Babel + LuaLaTeX), but I may be mistaken. As my document includes Hebrew, it is a bidirectional one.

Is there a way to make LuaLaTeX work in this case? Or how could I address the spacing issues otherwise?

Thank you.

**Comment by mcanouil** [comment]
You can modify the partials and/or template to fit your needs: https://quarto.org/docs/journals/templates.html#latex-partials

Mostly your questions are LaTeX questions, as such you are not in the best places for those which might already have answers elsewhere.

For advanced users, there are no workaround other than working with the underlying technology.
Quarto cannot offer abstraction there.


You might want to search with different keywords in other discussions, as the multi-language issue is not something new.
Some users reported how to mix English and Arabic for instance.

Edit: one of the discussions:

- https://github.com/quarto-dev/quarto-cli/discussions/6946

**Comment by cderv** [comment]
This is an example of a french document rendered with LuaLaTeX 
````markdown
---
title: Mon document
lang: fr
format:
  pdf:
    pdf-engine: lualatex
keep-tex: true
---

Ce document est en français
````

Babel is used from the pandoc's template with handling of LuaLaTeX and other 
````
\ifLuaTeX
\usepackage[bidi=basic]{babel}
\else
\usepackage[bidi=default]{babel}
\fi
\babelprovide[main,import]{french}
````

So this is possible. If you do a custom template, you need to write the correct LaTeX for this to work.

**Comment by normansimonr** [comment]
Hmm ok, I think this makes sense, thanks to both!
